### ISSUE TITLE
如何输出SPU NN的训练过程？

### ISSUE THREAD

**Issue by didadida321** [secretflow/secretflow#1091]
### Issue Type

Others

### Source

binary

### Secretflow Version

secretflow　

### OS Platform and Distribution

linux

### Python version

3.8

### What happend and What you expected to happen.

```shell
根据官网提供的NN例子，执行下面的代码是看不到训练进度的。有什么方式可以看到吗？因为数据集较大的时候，训练很久，没有日志输出，不知道是否在正常训练。
```

### Reproduction code to reproduce the issue.

```shell
params_spu = spu(train_auto_grad)(x1_, x2_, y_, init_params)
params = sf.reveal(params_spu)
print(params)
```

**Comment by tpppppub** [comment]
可以参考 SPU 里的这个[例子](https://github.com/secretflow/spu/blob/main/examples/python/ml/stax_nn/stax_nn.py#L54)，每次在 SPU 上运行一个 iteration。

**Comment by didadida321** [comment]
> 可以参考 SPU 里的这个[例子](https://github.com/secretflow/spu/blob/main/examples/python/ml/stax_nn/stax_nn.py#L54)，每次在 SPU 上运行一个 iteration。

这个例子和官网的似乎有些不一样。我理解应该是在train_auto_grad的body_fun增加打印epoch的日志，但是不知道如何添加，可以帮忙看看如何修改吗？
def train_auto_grad(x1, x2, y, params, n_batch=10, n_epochs=10, step_size=0.01):
    x = jnp.concatenate((x1, x2), axis=1)
    xs = jnp.array_split(x, len(x) / n_batch, axis=0)
    ys = jnp.array_split(y, len(y) / n_batch, axis=0)
    def body_fun(_, loop_carry):
        params = loop_carry
        for x, y in zip(xs, ys):
            _, grads = jax.value_and_grad(loss_func)(params, x, y)
            params = jax.tree_util.tree_map(
                lambda p, g: p - step_size * g, params, grads
            )
        return params

    params = jax.lax.fori_loop(0, n_epochs, body_fun, params)
    return params

**Comment by tpppppub** [comment]
因为 spu 并不是 eager 执行 train_auto_grad，而是将 train_auto_grad 编译后再执行，因此在 body_fun 里打印 epoch 日志是不行的

**Comment by didadida321** [comment]
> 因为 spu 并不是 eager 执行 train_auto_grad，而是将 train_auto_grad 编译后再执行，因此在 body_fun 里打印 epoch 日志是不行的

那基于官网的NN demo，有什么便捷的方式能看到训练进度吗？根据你发的例子进行修改，感觉改动挺大。
